tools/validate_spec.py

```diff
@@ -0,0 +1,314 @@
+        if self.count != 0:
+            self.file.write(",\n")
+        json.dump(
+            content,
+            self.file,
+            indent="\t",
+        )
+        self.count += 1
+
+    def close(self) -> None:
+        self.file.write("\n]")
+        self.file.close()
+
+
+class Validator:
+    def __init__(self, pdu_message: MessageValue) -> None:
+        self.pdu_message = pdu_message
+
+    def get_pdu_message(self) -> MessageValue:
+        return self.pdu_message.clone()
+
+    @staticmethod
+    def validate_message(
+        original_message: "Message", parser_result: "ParserResult"
+    ) -> "ValidationResult":
+
+        validation_result = ValidationResult(parser_result, original_message)
+        parsed_message = parser_result.get_parsed_message()
+        if isinstance(parsed_message, MessageValue):
+            validation_result.set_correct_serialization(
+                parsed_message.bytestring == original_message.bytes
+            )
+        else:
+            validation_result.set_correct_serialization(False)
+
+        if parser_result.is_valid and original_message.is_valid:
+            validation_result.set_classification(Classification.TP)
+        elif parser_result.is_valid and not original_message.is_valid:
+            validation_result.set_classification(Classification.FP)
+        elif not parser_result.is_valid and original_message.is_valid:
+            validation_result.set_classification(Classification.FN)
+        else:
+            validation_result.set_classification(Classification.TN)
+
+        return validation_result
+
+    def parse_message(self, message: bytes) -> "ParserResult":
+        pdu_model: MessageValue = self.get_pdu_message()
+        result = ParserResult()
+        try:
+            pdu_model.parse(message)
+            result.set_parsed_message(pdu_model)
+            # ISSUE: Componolit/RecordFlux#510
+        except Exception as e:  # pylint: disable=broad-except
+            result.set_error_message(str(e))
+        return result
+
+
+class Message:
+    def __init__(self, message_as_bytes: bytes, is_valid: bool, file_name: str):
+        self.bytes = message_as_bytes
+        self.is_valid = is_valid
+        self.file_name = file_name
+
+
+class ParserResult:
+    def __init__(self) -> None:
+        self._parsed_message: Optional[MessageValue] = None
+        self._error_message = ""
+        self._msg_valid: Optional[bool] = None
+
+    def set_parsed_message(self, parsed_message: MessageValue) -> None:
+        self._parsed_message = parsed_message
+        self._msg_valid = parsed_message.valid_message
+
+    def set_error_message(self, error_message: str) -> None:
+        self._error_message = error_message
+        self._msg_valid = False
+
+    def get_parsed_message(self) -> Optional[MessageValue]:
+        return self._parsed_message
+
+    def get_error_message(self) -> str:
+        return self._error_message
```

Comment:
Python properties should be used instead of getter/setter methods.

Reply:
Changed in 7a4baa8